keras_hub/src/models/retinanet/retinanet_backbone_test.py

```diff
@@ -0,0 +1,53 @@
+import pytest
+from keras import ops
+
+from keras_hub.src.models.resnet.resnet_backbone import ResNetBackbone
+from keras_hub.src.models.retinanet.retinanet_backbone import RetinaNetBackbone
+from keras_hub.src.tests.test_case import TestCase
+
+
+class RetinaNetBackboneTest(TestCase):
+    def setUp(self):
+        resnet_kwargs = {
+            "input_conv_filters": [64],
+            "input_conv_kernel_sizes": [7],
+            "stackwise_num_filters": [64, 64, 64],
+            "stackwise_num_blocks": [2, 2, 2],
+            "stackwise_num_strides": [1, 2, 2],
+            "image_shape": (None, None, 3),
+            "block_type": "bottleneck_block",
+            "use_pre_activation": False,
+        }
+        image_encoder = ResNetBackbone(**resnet_kwargs)
+
+        self.init_kwargs = {
+            "image_encoder": image_encoder,
+            "min_level": 3,
+            "max_level": 4,
+        }
+
+        self.input_size = 256
+        self.input_data = ops.ones((2, self.input_size, self.input_size, 3))
+
+    def test_backbone_basics(self):
+        self.run_vision_backbone_test(
+            cls=RetinaNetBackbone,
+            init_kwargs=self.init_kwargs,
+            input_data=self.input_data,
+            expected_output_shape={
+                "P3": (2, 32, 32, 256),
+                "P4": (2, 16, 16, 256),
+            },
+            expected_pyramid_output_keys=False,
```

Comment:
this needs to be enabled

Reply:
![image](https://github.com/user-attachments/assets/c52b2c70-d084-4453-b43a-3a3ff5aade85)

If you see the test case it is written for standalone backbones. Here they create a model with expected outputs to be same as `backbone.pyramid_outputs`. But in retinanet that's not the case it has one more layer where `fpn` layer has control over how many can be in output. Standard RetinaNet starts from `"P3"`.

Reply:
Making some changes to `RetinaNetbackbone` to make it compatible.